src/components/ItemGrid.vue

```diff
@@ -0,0 +1,24 @@
+<template>
+  <div class="row">
+    <app-item v-for="(item, index) in items" @click.native="deleteItem(index)">{{ item }}</app-item>
```

Comment:
Když používáš for, je potřeba dodat i `:key="index"`
Na to by tě upozornil linter.
Tohle by mohlo i fungovat ve vue automaticky. Nikdy jsem nepřeboval nijak extra indexovat